### PR TITLE
Add Mark all read Button

### PR DESCRIPTION
**Pull request by jiaqihuang-233**
1. Add a mark all as read button
2. Show error when invitations fail

### CODE REVIEW

File: client/components/InviteModal/InviteModal.tsx [hunk 1]
```diff
@@ -7,6 +7,18 @@ import { toast } from 'react-toastify';
 
 export default ({ onExit, document }) => {
   const client = useSelector(getClient);
+  let success = true;
```
Comment (lbittner-pdftron) [inline client/components/InviteModal/InviteModal.tsx:10]: Hey Jackie, this is not a good way to do this as every time this function is rendered this variable will be reset to `true`. Use a ref or state instead

File: client/components/InviteModal/InviteModal.tsx [hunk 2]
```diff
@@ -16,7 +28,11 @@ export default ({ onExit, document }) => {
         return;
       }
       await client.inviteUsersToDocument(document.id, list);
-      toast.success('Success!');
+      if (success) {
+        toast.success('Success!');
+      } else {
+        toast.error('Error: No permission or user is already invited.');
+      }
```
Comment (lbittner-pdftron) [inline client/components/InviteModal/InviteModal.tsx:35]: You shouldn't be relying on the `permissionError` callback to check if this worked. Doesn't `inviteUsersToDocument` throw an error if there are no permissions?

**Review by lbittner-pdftron — CHANGES_REQUESTED** [review]

File: client/components/InviteModal/InviteModal.tsx (hunk 2 above)
Comment (jiaqihuang-233) [reply client/components/InviteModal/InviteModal.tsx:35]: Hey @logan, no it doesn't. It only has this line `this.trigger('permissionError', 'Document', 'invite');` in the catch block.

Comment (lbittner-pdftron) [reply client/components/InviteModal/InviteModal.tsx:35]: Ok, I think we should fix that to actually throw an error then

**Review by lbittner-pdftron — APPROVED** [review]

**Pull request merged**